uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/src/main/java/org/uberfire/ext/security/management/wildfly/properties/WildflyGroupPropertiesManager.java

```diff
@@ -101,24 +101,41 @@ public Group get(String identifier) throws SecurityManagementException {
         throw new GroupNotFoundException(identifier);
     }
 
-    public Set<Group> getGroupsForUser(String username) {
+    public Set[] getGroupsAndRolesForUser(String username) {
         if (groupsPropertiesFileLoader != null && username != null) {
             try {
                 final String groupsStr = groupsPropertiesFileLoader.getProperties().getProperty(username);
                 final Set<String> groups = parseGroupIdentifiers(groupsStr);
+                final Set<String> registeredRoles = SecurityManagementUtils.getRegisteredRoleNames();
                 if (groups != null) {
                     final Set<String> allGroups = getAllGroups();
                     if (allGroups != null) {
-                        final Set<Group> result = new HashSet<Group>(groups.size());
+                        final Set<Group> _groups = new HashSet<Group>();
+                        final Set<Role> _roles = new HashSet<Role>();
                         for (final String name : groups) {
                             if (!allGroups.contains(name)) {
                                 String error = "Error getting groups for user. User's group '" + name + "' does not exist.";
                                 LOG.error(error);
                                 throw new SecurityManagementException(error);
                             }
-                            result.add(createGroup(name));
+                            
+                            if (registeredRoles.contains(name)) {
+                                // Is a role.
```

Comment:
Don't like that this code is duplicated. Can we move it to a base class or reuse it somehow?